ksql-engine/src/main/java/io/confluent/ksql/util/ExpressionTypeManager.java

```diff
@@ -246,4 +252,34 @@ private Schema resolveArithmeticType(final Schema leftSchema,
                                        final Schema rightSchema) {
     return SchemaUtil.resolveArithmeticType(leftSchema.type(), rightSchema.type());
   }
+
+  private static void validateComparisonOperandTypes(
+      final ComparisonExpression.Type operator,
+      final Schema leftType,
+      final Schema rightType) {
```

Comment:
can we write this method to accept N types instead of just 2? I'm thinking about leveraging this for the BETWEEN operator PR 😂

Reply:
Hmm, that's gonna be tricky to have a concise (and not too ugly) implementation for the function. Do you have any suggestion?

Reply:
just noticed that you responded to this - I can take a stab at it after you merge